### ISSUE TITLE
Setting MaxConcurrentWorkflowTaskExecutionSize to 1 results in worker not executing any tasks

### ISSUE THREAD

**Issue by phudtran** [temporalio/sdk-go#1003]
## Expected Behavior

Worker executes 1 workflow at a time.

## Actual Behavior

Worker does nothing

## Steps to Reproduce the Problem

  1. Create a new worker
  1. Set MaxConcurrentWorkflowTaskExecutionSize to 1 in the worker options

## Specifications

  - Version: v1.19.0
  - Platform: Ubuntu 22.04.1 LTS (GNU/Linux 5.15.0-1026-aws x86_64)

**Comment by dnr** [comment]
Sorry, I confused this setting with workflow task _pollers_ (which does have min 2) and mentioned it to Yimin, so that was my mistake. As far as I know this limitation is not documented or known.

**Comment by phudtran** [comment]
I noticed the other documented settings had this limitation so I tried setting MaxConcurrentWorkflowTaskExecutionSize to 2. It seems to do the correct behavior in my tests last week :grinning:

**Comment by dnr** [comment]
(Btw limiting concurrent workflow task execution is probably not the right solution for anything. If you share why you wanted to do this you might get a better alternative.)

**Comment by Quinn-With-Two-Ns** [comment]
Looks like we limit the number of pollers to `MaxConcurrentWorkflowTaskExecutionSize ` so the worker will only be polling on the sticky queue, so it will never pick up work. So the root cause is the same reason we need 2 workflow task pollers

Note: Technically we do create `maxConcurrentWorkflowTaskPollers ` number of pollers , but only really start `MaxConcurrentWorkflowTaskExecutionSize `

`MaxConcurrentWorkflowTaskExecutionSize` should never be less than `MaxConcurrentWorkflowTaskPollers `

We call this out here https://docs.temporal.io/application-development/worker-performance#invariants but say it only applies to java which is not true it also applies to the go sdk from reading the code.

Basically we need to document this and probably force `MaxConcurrentWorkflowTaskExecutionSize` to be greater than `MaxConcurrentWorkflowTaskPollers `

**Comment by phudtran** [comment]
> (Btw limiting concurrent workflow task execution is probably not the right solution for anything. If you share why you wanted to do this you might get a better alternative.)

We're in the early stages of using Temporal for a cloud simulation project.
In our case, we're using workflows to represent a _set of VMs_ to be created on a Node, and activities representing each of the VMs to be created. I only stumbled upon this MaxConcurrentWorkflowTaskExecutionSize behavior because of a temporary accommodation for an upstream service in test.
